### ISSUE TITLE
Event Overview filter broken in some cases

### ISSUE THREAD

**Issue by bobapple** [Icinga/icingaweb2#3848]
Using the following filters within the "Event Overview" leads to an error:

* `service`
* `servicegroup_alias`
* `servicegroup`
* `host`
* `hostgroup`
* `hostgroup_alias`

I get the following error message when using one of those filter:

`Icinga\Module\Monitoring\Backend\Ido\Query\HostnotificationQuery : Got invalid column: host`

If I use the `output` filter, I get the following error message:

```
SQLSTATE[HY000]: General error: 1111 Invalid use of group function, query was: SELECT eh.id, eh.host_name, eh.host_display_name, eh.service_description, eh.service_display_name, eh.object_type, eh.timestamp, eh.state, eh.output, eh.type FROM (SELECT sth.id, sth.timestamp, sth.output, sth.type, sth.state, sth.object_type, sth.host_name, sth.service_description, sth.host_display_name, sth.service_display_name FROM (SELECT hh.statehistory_id AS id, ('host') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, ho.name1 AS host_name, NULL AS service_description, NULL AS service_display_name, NULL AS service_host_name, (CASE WHEN hh.state_type = 1 THEN 'hard_state' ELSE 'soft_state' END) AS type, UNIX_TIMESTAMP(hh.state_time) AS timestamp, hh.object_id, hh.state, (CASE WHEN hh.state_type = 1 THEN hh.output ELSE '[ ' || hh.current_check_attempt || '/' || hh.max_check_attempts || ' ] ' || hh.output END) AS output FROM icinga_statehistory AS hh
INNER JOIN icinga_objects AS ho ON ho.object_id = hh.object_id AND ho.is_active = 1 AND ho.objecttype_id = 1
INNER JOIN icinga_hosts AS h ON h.host_object_id = ho.object_id WHERE ((CASE WHEN hh.state_type = 1 THEN hh.output ELSE '[ ' || hh.current_check_attempt || '/' || hh.max_check_attempts || ' ] ' || hh.output END) = 'foobar') UNION ALL SELECT sh.statehistory_id AS id, ('service') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, so.name1 AS host_name, so.name2 AS service_description, s.display_name COLLATE latin1_general_ci AS service_display_name, so.name1 AS service_host_name, (CASE WHEN sh.state_type = 1 THEN 'hard_state' ELSE 'soft_state' END) AS type, UNIX_TIMESTAMP(sh.state_time) AS timestamp, sh.object_id, sh.state, (CASE WHEN sh.state_type = 1 THEN sh.output ELSE '[ ' || sh.current_check_attempt || '/' || sh.max_check_attempts || ' ] ' || sh.output END) AS output FROM icinga_statehistory AS sh
INNER JOIN icinga_objects AS so ON so.object_id = sh.object_id AND so.is_active = 1 AND so.objecttype_id = 2
INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id
INNER JOIN icinga_hosts AS h ON h.host_object_id = s.host_object_id WHERE ((CASE WHEN sh.state_type = 1 THEN sh.output ELSE '[ ' || sh.current_check_attempt || '/' || sh.max_check_attempts || ' ] ' || sh.output END) = 'foobar')) AS sth UNION ALL SELECT dsh.id, dsh.timestamp, dsh.output, dsh.type, dsh.state, dsh.object_type, dsh.host_name, dsh.service_description, dsh.host_display_name, dsh.service_display_name FROM (SELECT hdh.downtimehistory_id AS id, ('host') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, ho.name1 AS host_name, NULL AS service_description, NULL AS service_display_name, NULL AS service_host_name, ('dt_start') AS type, UNIX_TIMESTAMP(hdh.actual_start_time) AS timestamp, hdh.object_id, (-1) AS state, ('[' || hdh.author_name || '] ' || hdh.comment_data) AS output FROM icinga_downtimehistory AS hdh
INNER JOIN icinga_objects AS ho ON ho.object_id = hdh.object_id AND ho.is_active = 1 AND ho.objecttype_id = 1
INNER JOIN icinga_hosts AS h ON h.host_object_id = ho.object_id WHERE (hdh.actual_start_time > '1970-01-02 00:00:00') AND (('[' || hdh.author_name || '] ' || hdh.comment_data) = 'foobar') UNION ALL SELECT sdh.downtimehistory_id AS id, ('service') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, so.name1 AS host_name, so.name2 AS service_description, s.display_name COLLATE latin1_general_ci AS service_display_name, so.name1 AS service_host_name, ('dt_start') AS type, UNIX_TIMESTAMP(sdh.actual_start_time) AS timestamp, sdh.object_id, (-1) AS state, ('[' || sdh.author_name || '] ' || sdh.comment_data) AS output FROM icinga_downtimehistory AS sdh
INNER JOIN icinga_objects AS so ON so.object_id = sdh.object_id AND so.is_active = 1 AND so.objecttype_id = 2
INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id
INNER JOIN icinga_hosts AS h ON h.host_object_id = s.host_object_id WHERE (sdh.actual_start_time > '1970-01-02 00:00:00') AND (('[' || sdh.author_name || '] ' || sdh.comment_data) = 'foobar')) AS dsh UNION ALL SELECT deh.id, deh.timestamp, deh.output, deh.type, deh.state, deh.object_type, deh.host_name, deh.service_description, deh.host_display_name, deh.service_display_name FROM (SELECT hdh.downtimehistory_id AS id, ('host') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, ho.name1 AS host_name, NULL AS service_description, NULL AS service_display_name, NULL AS service_host_name, ('dt_end') AS type, UNIX_TIMESTAMP(hdh.actual_end_time) AS timestamp, hdh.object_id, (-1) AS state, ('[' || hdh.author_name || '] ' || hdh.comment_data) AS output FROM icinga_downtimehistory AS hdh
INNER JOIN icinga_objects AS ho ON ho.object_id = hdh.object_id AND ho.is_active = 1 AND ho.objecttype_id = 1
INNER JOIN icinga_hosts AS h ON h.host_object_id = ho.object_id WHERE (hdh.actual_end_time > '1970-01-02 00:00:00') AND (('[' || hdh.author_name || '] ' || hdh.comment_data) = 'foobar') UNION ALL SELECT sdh.downtimehistory_id AS id, ('service') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, so.name1 AS host_name, so.name2 AS service_description, s.display_name COLLATE latin1_general_ci AS service_display_name, so.name1 AS service_host_name, ('dt_end') AS type, UNIX_TIMESTAMP(sdh.actual_end_time) AS timestamp, sdh.object_id, (-1) AS state, ('[' || sdh.author_name || '] ' || sdh.comment_data) AS output FROM icinga_downtimehistory AS sdh
INNER JOIN icinga_objects AS so ON so.object_id = sdh.object_id AND so.is_active = 1 AND so.objecttype_id = 2
INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id
INNER JOIN icinga_hosts AS h ON h.host_object_id = s.host_object_id WHERE (sdh.actual_end_time > '1970-01-02 00:00:00') AND (('[' || sdh.author_name || '] ' || sdh.comment_data) = 'foobar')) AS deh UNION ALL SELECT ch.id, ch.timestamp, ch.output, ch.type, ch.state, ch.object_type, ch.host_name, ch.service_description, ch.host_display_name, ch.service_display_name FROM (SELECT hch.commenthistory_id AS id, ('host') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, ho.name1 AS host_name, NULL AS service_description, NULL AS service_display_name, NULL AS service_host_name, (CASE hch.entry_type WHEN 1 THEN 'comment' WHEN 2 THEN 'dt_comment' WHEN 3 THEN 'flapping' WHEN 4 THEN 'ack' END) AS type, UNIX_TIMESTAMP(hch.comment_time) AS timestamp, hch.object_id, (-1) AS state, ('[' || hch.author_name || '] ' || hch.comment_data) AS output FROM icinga_commenthistory AS hch
INNER JOIN icinga_objects AS ho ON ho.object_id = hch.object_id AND ho.is_active = 1 AND ho.objecttype_id = 1
INNER JOIN icinga_hosts AS h ON h.host_object_id = ho.object_id WHERE (('[' || hch.author_name || '] ' || hch.comment_data) = 'foobar') UNION ALL SELECT sch.commenthistory_id AS id, ('service') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, so.name1 AS host_name, so.name2 AS service_description, s.display_name COLLATE latin1_general_ci AS service_display_name, so.name1 AS service_host_name, (CASE sch.entry_type WHEN 1 THEN 'comment' WHEN 2 THEN 'dt_comment' WHEN 3 THEN 'flapping' WHEN 4 THEN 'ack' END) AS type, UNIX_TIMESTAMP(sch.comment_time) AS timestamp, sch.object_id, (-1) AS state, ('[' || sch.author_name || '] ' || sch.comment_data) AS output FROM icinga_commenthistory AS sch
INNER JOIN icinga_objects AS so ON so.object_id = sch.object_id AND so.is_active = 1 AND so.objecttype_id = 2
INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id
INNER JOIN icinga_hosts AS h ON h.host_object_id = s.host_object_id WHERE (('[' || sch.author_name || '] ' || sch.comment_data) = 'foobar') GROUP BY sch.commenthistory_id,
so.object_id,
s.service_id) AS ch UNION ALL SELECT cdh.id, cdh.timestamp, cdh.output, cdh.type, cdh.state, cdh.object_type, cdh.host_name, cdh.service_description, cdh.host_display_name, cdh.service_display_name FROM (SELECT hch.commenthistory_id AS id, ('host') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, ho.name1 AS host_name, NULL AS service_description, NULL AS service_display_name, NULL AS service_host_name, (CASE hch.entry_type WHEN 1 THEN 'comment' WHEN 2 THEN 'dt_comment' WHEN 3 THEN 'flapping' WHEN 4 THEN 'ack' END || '_deleted') AS type, UNIX_TIMESTAMP(hch.deletion_time) AS timestamp, hch.object_id, (-1) AS state, ('[' || hch.author_name || '] ' || hch.comment_data) AS output FROM icinga_commenthistory AS hch
INNER JOIN icinga_objects AS ho ON ho.object_id = hch.object_id AND ho.is_active = 1 AND ho.objecttype_id = 1
INNER JOIN icinga_hosts AS h ON h.host_object_id = ho.object_id WHERE (hch.deletion_time > '1970-01-02 00:00:00') AND (('[' || hch.author_name || '] ' || hch.comment_data) = 'foobar') UNION ALL SELECT sch.commenthistory_id AS id, ('service') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, so.name1 AS host_name, so.name2 AS service_description, s.display_name COLLATE latin1_general_ci AS service_display_name, so.name1 AS service_host_name, (CASE sch.entry_type WHEN 1 THEN 'comment' WHEN 2 THEN 'dt_comment' WHEN 3 THEN 'flapping' WHEN 4 THEN 'ack' END || '_deleted') AS type, UNIX_TIMESTAMP(sch.deletion_time) AS timestamp, sch.object_id, (-1) AS state, ('[' || sch.author_name || '] ' || sch.comment_data) AS output FROM icinga_commenthistory AS sch
INNER JOIN icinga_objects AS so ON so.object_id = sch.object_id AND so.is_active = 1 AND so.objecttype_id = 2
INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id
INNER JOIN icinga_hosts AS h ON h.host_object_id = s.host_object_id WHERE (sch.deletion_time > '1970-01-02 00:00:00') AND (('[' || sch.author_name || '] ' || sch.comment_data) = 'foobar') GROUP BY sch.commenthistory_id,
so.object_id,
s.service_id) AS cdh UNION ALL SELECT n.id, n.timestamp, n.output, ('notify') AS type, n.state, n.object_type, n.host_name, n.service_description, n.host_display_name, n.service_display_name FROM (SELECT hn.notification_id AS id, UNIX_TIMESTAMP(hn.start_time) AS timestamp, ('[' || GROUP_CONCAT(DISTINCT co.name1 ORDER BY co.name1 SEPARATOR ', ') COLLATE latin1_general_ci || '] ' || hn.output) AS output, hn.state, ('host') AS object_type, ho.name1 AS host_name, NULL AS service_description, h.display_name COLLATE latin1_general_ci AS host_display_name, NULL AS service_display_name FROM icinga_notifications AS hn
INNER JOIN icinga_objects AS ho ON ho.object_id = hn.object_id AND ho.is_active = 1 AND ho.objecttype_id = 1
LEFT JOIN icinga_contactnotifications AS cn ON cn.notification_id = hn.notification_id
LEFT JOIN icinga_objects AS co ON co.object_id = cn.contact_object_id AND co.is_active = 1 AND co.objecttype_id = 10
INNER JOIN icinga_hosts AS h ON h.host_object_id = ho.object_id WHERE (('[' || GROUP_CONCAT(DISTINCT co.name1 ORDER BY co.name1 SEPARATOR ', ') COLLATE latin1_general_ci || '] ' || hn.output) = 'foobar') GROUP BY hn.notification_id,
ho.object_id,
h.host_id UNION ALL SELECT sn.notification_id AS id, UNIX_TIMESTAMP(sn.start_time) AS timestamp, ('[' || GROUP_CONCAT(DISTINCT co.name1 ORDER BY co.name1 SEPARATOR ', ') COLLATE latin1_general_ci || '] ' || sn.output) AS output, sn.state, ('service') AS object_type, so.name1 AS host_name, so.name2 AS service_description, h.display_name COLLATE latin1_general_ci AS host_display_name, s.display_name COLLATE latin1_general_ci AS service_display_name FROM icinga_notifications AS sn
INNER JOIN icinga_objects AS so ON so.object_id = sn.object_id AND so.is_active = 1 AND so.objecttype_id = 2
LEFT JOIN icinga_contactnotifications AS cn ON cn.notification_id = sn.notification_id
LEFT JOIN icinga_objects AS co ON co.object_id = cn.contact_object_id AND co.is_active = 1 AND co.objecttype_id = 10
INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id
INNER JOIN icinga_hosts AS h ON h.host_object_id = s.host_object_id WHERE (('[' || GROUP_CONCAT(DISTINCT co.name1 ORDER BY co.name1 SEPARATOR ', ') COLLATE latin1_general_ci || '] ' || sn.output) = 'foobar') GROUP BY sn.notification_id,
so.object_id,
h.host_id,
s.service_id) AS n UNION ALL SELECT fsh.id, fsh.timestamp, fsh.output, fsh.type, fsh.state, fsh.object_type, fsh.host_name, fsh.service_description, fsh.host_display_name, fsh.service_display_name FROM (SELECT hfh.flappinghistory_id AS id, ('host') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, ho.name1 AS host_name, NULL AS service_description, NULL AS service_display_name, NULL AS service_host_name, ('flapping') AS type, UNIX_TIMESTAMP(hfh.event_time) AS timestamp, hfh.object_id, (-1) AS state, (hfh.percent_state_change || '') AS output FROM icinga_flappinghistory AS hfh
INNER JOIN icinga_objects AS ho ON ho.object_id = hfh.object_id AND ho.is_active = 1 AND ho.objecttype_id = 1
INNER JOIN icinga_hosts AS h ON h.host_object_id = ho.object_id WHERE (hfh.event_type = 1000) AND ((hfh.percent_state_change || '') = 'foobar') UNION ALL SELECT sfh.flappinghistory_id AS id, ('service') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, so.name1 AS host_name, so.name2 AS service_description, s.display_name COLLATE latin1_general_ci AS service_display_name, so.name1 AS service_host_name, ('flapping') AS type, UNIX_TIMESTAMP(sfh.event_time) AS timestamp, sfh.object_id, (-1) AS state, (sfh.percent_state_change || '') AS output FROM icinga_flappinghistory AS sfh
INNER JOIN icinga_objects AS so ON so.object_id = sfh.object_id AND so.is_active = 1 AND so.objecttype_id = 2
INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id
INNER JOIN icinga_hosts AS h ON h.host_object_id = s.host_object_id WHERE (sfh.event_type = 1000) AND ((sfh.percent_state_change || '') = 'foobar')) AS fsh UNION ALL SELECT fsh.id, fsh.timestamp, fsh.output, fsh.type, fsh.state, fsh.object_type, fsh.host_name, fsh.service_description, fsh.host_display_name, fsh.service_display_name FROM (SELECT hfh.flappinghistory_id AS id, ('host') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, ho.name1 AS host_name, NULL AS service_description, NULL AS service_display_name, NULL AS service_host_name, ('flapping_deleted') AS type, UNIX_TIMESTAMP(hfh.event_time) AS timestamp, hfh.object_id, (-1) AS state, (hfh.percent_state_change || '') AS output FROM icinga_flappinghistory AS hfh
INNER JOIN icinga_objects AS ho ON ho.object_id = hfh.object_id AND ho.is_active = 1 AND ho.objecttype_id = 1
INNER JOIN icinga_hosts AS h ON h.host_object_id = ho.object_id WHERE (hfh.event_type = 1001) AND ((hfh.percent_state_change || '') = 'foobar') UNION ALL SELECT sfh.flappinghistory_id AS id, ('service') AS object_type, h.display_name COLLATE latin1_general_ci AS host_display_name, so.name1 AS host_name, so.name2 AS service_description, s.display_name COLLATE latin1_general_ci AS service_display_name, so.name1 AS service_host_name, ('flapping_deleted') AS type, UNIX_TIMESTAMP(sfh.event_time) AS timestamp, sfh.object_id, (-1) AS state, (sfh.percent_state_change || '') AS output FROM icinga_flappinghistory AS sfh
INNER JOIN icinga_objects AS so ON so.object_id = sfh.object_id AND so.is_active = 1 AND so.objecttype_id = 2
INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id
INNER JOIN icinga_hosts AS h ON h.host_object_id = s.host_object_id WHERE (sfh.event_type = 1001) AND ((sfh.percent_state_change || '') = 'foobar')) AS fsh) AS eh ORDER BY eh.timestamp DESC LIMIT 25

SQLSTATE[HY000]: General error: 1111 Invalid use of group function
```

**Comment by nilmerg** [comment]
Which version of Icinga Web are we talking about here? That `host` and `service` is not found should be fixed (#3721) since v2.6.3.

The other ones are indeed new.

**Comment by nilmerg** [comment]
`output` is fixed since #2266 in the current master.
